sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_models.py

```diff
@@ -2,11 +2,363 @@
+    """Maps an `AsymmetricPadding` to an encryption algorithm.
+
+    :param padding: The padding to use.
+    :type padding: AsymmetricPadding
+
+    :returns: The corresponding Key Vault encryption algorithm.
+    :rtype: EncryptionAlgorithm
+    """
+    if isinstance(padding, OAEP):
+        # Public algorithm property was only added in https://github.com/pyca/cryptography/pull/9582
+        # _algorithm property has been available in every version of the OAEP class, so we use it as a backup
+        try:
+            algorithm = padding.algorithm  # type: ignore[attr-defined]
+        except AttributeError:
+            algorithm = padding._algorithm  # pylint:disable=protected-access
+        mapped_algorithm = OAEP_MAP.get(type(algorithm))
+        if mapped_algorithm is None:
+            raise ValueError(f"Unsupported algorithm: {algorithm.name}")
+
+        # Public mgf property was added at the same time as algorithm
+        try:
+            mgf = padding.mgf  # type: ignore[attr-defined]
+        except AttributeError:
+            mgf = padding._mgf  # pylint:disable=protected-access
+        if not isinstance(mgf, MGF1):
+            raise ValueError(f"Unsupported MGF: {mgf}")
+
+    elif isinstance(padding, PKCS1v15):
+        mapped_algorithm = EncryptionAlgorithm.rsa1_5
+    else:
+        raise ValueError(f"Unsupported padding: {padding.name}")
+
+    return mapped_algorithm
+
+
+def get_signature_algorithm(padding: AsymmetricPadding, algorithm: HashAlgorithm) -> SignatureAlgorithm:
+    """Maps an `AsymmetricPadding` and `HashAlgorithm` to a signature algorithm.
+
+    :param padding: The padding to use.
+    :type padding: AsymmetricPadding
+    :param algorithm: The algorithm to use.
+    :type algorithm: HashAlgorithm
+
+    :returns: The corresponding Key Vault signature algorithm.
+    :rtype: SignatureAlgorithm
+    """
+    mapped_algorithm = SIGN_ALGORITHM_MAP.get(type(algorithm))
+    if mapped_algorithm is None:
+        raise ValueError(f"Unsupported algorithm: {algorithm.name}")
+
+    # If PSS padding is requested, use the PSS equivalent algorithm
+    if isinstance(padding, PSS):
+        mapped_algorithm = PSS_MAP.get(mapped_algorithm)
+
+        # Public mgf property was only added in https://github.com/pyca/cryptography/pull/9582
+        # _mgf property has been available in every version of the PSS class, so we use it as a backup
+        try:
+            mgf = padding.mgf  # type: ignore[attr-defined]
+        except AttributeError:
+            mgf = padding._mgf  # pylint:disable=protected-access
+        if not isinstance(mgf, MGF1):
+            raise ValueError(f"Unsupported MGF: {mgf}")
+
+    # The only other padding accepted is PKCS1v15
+    elif not isinstance(padding, PKCS1v15):
+        raise ValueError(f"Unsupported padding: {padding.name}")
+
+    return cast(SignatureAlgorithm, mapped_algorithm)
+
+
+class KeyVaultRSAPublicKey(RSAPublicKey):
```

Comment:
How do we feel about async implementations?
I guess there's no async protocol for these keys.... so there's not really a fit for it...
I guess it can always be added at a later date if customer demand or underlying support is added.

Though maybe we should specify in the docstring that only synchronous clients/operations are supported?